### PR TITLE
Added get_shape method

### PR DESCRIPTION
**Pull request by Lumijek**
## PR Summary
Added `get_shape` method and updated `__str__` to AxesImage as specified by (#22494)
## PR Checklist

**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A ] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/image.py [hunk 1]
```diff
@@ -293,6 +293,15 @@ def get_size(self):
 
         return self._A.shape[:2]
 
+    def get_shape(self):
+        """Return the shape of the image as
+        tuple (numrows, numcols, channels).
+        """
```
Comment (oscargus) [inline lib/matplotlib/image.py:299]:
```suggestion
        """
        Return the shape of the image as tuple (numrows, numcols, channels).
        """
```

**Comment by oscargus** [comment]
Please add tests for these changes.

You can example generate a test image as:
```
    ax = plt.gca()
    im = AxesImage(ax)
    z = np.arange(12, dtype=float).reshape((4, 3))
    im.set_data(z)
```
and then call the methods (in separate tests) and add an `assert` for the expected results. Add these tests to https://github.com/matplotlib/matplotlib/blob/main/lib/matplotlib/tests/test_image.py

File: lib/matplotlib/image.py (hunk 1 above)
Comment (tacaswell) [reply lib/matplotlib/image.py:299]: Can you use the numpydoc format for the returns?

**Comment by tacaswell** [comment]
Please do not merge the main branch into your feature branch (we prefer to rebase if there are conflicts).

File: lib/matplotlib/image.py (hunk 1 above)
Comment (Lumijek) [reply lib/matplotlib/image.py:299]: I was going to do that but line length exceeded pep8 limit.

**Comment by timhoffm** [comment]
@Lumijek are you still planning to continue on this? If not, I'd pick this up so that it can move forward.

**Comment by jklymak** [comment]
Moved to draft until someone picks this up.

**Comment by saranti** [comment]
It's been a while so I can write some tests for this. I don't know what the procedure is for picking up orphaned PRs though.

**Comment by rcomer** [comment]
Thanks @saranti, basically just create a new branch and a new PR referencing this one.  You could `git cherry-pick` the relevant commit from here into your branch, or just write your new branch from scratch, whichever you prefer.

**Comment by tacaswell** [comment]
Closing in favor of #25425

**Pull request closed without merging**